### ISSUE TITLE
Installing fails because chart museum cannot be reached

### ISSUE THREAD

**Issue by i386** [jenkins-x/jx#219]
*Error*
```
waiting for install to be ready, if this is the first time then it will take a while to download images
Error: Looks like "http://admin:admin@chartmuseum.default.192.168.64.7.nip.io" is not a valid chart repository or cannot be reached: Get http://admin:admin@chartmuseum.default.192.168.64.7.nip.io/index.yaml: dial tcp 192.168.64.7:80: getsockopt: connection refused
Error: Command failed  helm repo add releases http://admin:admin@chartmuseum.default.192.168.64.7.nip.io
```

Result of `jx version`:
```
pounder:~ jdumay$ jx version
WARNING: Failed to find draft version: exit status 1
WARNING: Client: &version.Version{SemVer:"v0.10.1", GitCommit:"486291360aeffa2fe0a131fbf0b4f3c601056b16", GitTreeState:"clean"}
2018/02/08 13:26:14 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp: missing address"; Reconnecting to { <nil>}
2018/02/08 13:26:15 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp: missing address"; Reconnecting to { <nil>}
2018/02/08 13:26:17 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp: missing address"; Reconnecting to { <nil>}
Error: cannot connect to draftd
NAME        VERSION                    
jx          1.0.53            
Jenkins X   0.0.55            
Kubernetes  v1.9.0            
Helm Client v2.8.0+g14af25f   
Helm Server v2.8.0+g14af25f   
Git         git version 2.16.1
```

*Full log*
```
pounder:~ jdumay$ jx uninstall; jx install
? Are you sure you wish to remove the Jenkins X platform from the 'default' namespace on cluster 'https://192.168.64.7:8443'? : Yes
release "jenkins-x" deleted
Jenkins X has been successfully uninstalled 
Lets set up a git username and API token to be able to perform CI / CD

? Do you wish to use i386 as the GitHub username for CI/CD pipelines: Yes


You can now configure a wildcard DNS pointing to the new loadbalancer address 192.168.64.7
? Domain 192.168.64.7.nip.io
Cloning the Jenkins X cloud environments repo to /Users/jdumay/.jx/cloud-environments
? A local Jenkins X cloud environments repository already exists, recreate with latest? Yes
Cloning the Jenkins X cloud environments repo to /Users/jdumay/.jx/cloud-environments
Counting objects: 239, done.
Compressing objects: 100% (25/25), done.
Total 239 (delta 12), reused 28 (delta 7), pack-reused 205
helm repo add jenkins-x http://chartmuseum.build.cd.jenkins-x.io
"jenkins-x" has been added to your repositories
rm -rf secrets.yaml.dec
helm repo add jenkins-x http://chartmuseum.build.cd.jenkins-x.io
"jenkins-x" has been added to your repositories
helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "jenkins-x" chart repository
...Successfully got an update from the "incubator" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 
helm install jenkins-x/jenkins-x-platform --name jenkins-x -f ./myvalues.yaml -f ./secrets.yaml --version 0.0.55 --values=/Users/jdumay/.jx/gitSecrets.yaml --values=/Users/jdumay/.jx/extraValues.yaml --namespace=default
NAME:   jenkins-x
LAST DEPLOYED: Thu Feb  8 13:03:13 2018
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1/ServiceAccount
NAME                SECRETS  AGE
cleanup             1        32s
expose              1        32s
pipelinecontroller  1        32s

==> v1/Service
NAME                            TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)    AGE
jenkins-x-chartmuseum           ClusterIP  10.102.116.155  <none>       8080/TCP   32s
jenkins-x-docker-registry       ClusterIP  10.105.56.116   <none>       5000/TCP   32s
heapster                        ClusterIP  10.99.17.32     <none>       8082/TCP   32s
jenkins-agent                   ClusterIP  10.108.54.139   <none>       50000/TCP  32s
jenkins                         ClusterIP  10.103.220.210  <none>       8080/TCP   32s
jenkins-x-kubernetes-dashboard  ClusterIP  10.98.8.147     <none>       80/TCP     32s
jenkins-x-mongodb               ClusterIP  10.96.97.148    <none>       27017/TCP  32s
jenkins-x-monocular-api         ClusterIP  10.107.106.179  <none>       80/TCP     32s
jenkins-x-monocular-prerender   ClusterIP  10.103.210.5    <none>       80/TCP     32s
jenkins-x-monocular-ui          ClusterIP  10.106.230.62   <none>       80/TCP     32s
jenkins-x-sonatype-nexus        ClusterIP  10.105.254.76   <none>       8081/TCP   32s

==> v1beta1/Deployment
NAME                            DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
jenkins-x-chartmuseum           1        1        1           1          32s
jenkins-x-docker-registry       1        1        1           1          32s
jenkins-x-heapster              1        1        1           1          32s
jenkins                         1        1        1           0          32s
jenkins-x-kubernetes-dashboard  1        1        1           1          32s
jenkins-x-mongodb               1        1        1           1          32s
jenkins-x-monocular-api         1        1        1           0          32s
jenkins-x-monocular-prerender   1        1        1           1          32s
jenkins-x-monocular-ui          1        1        1           0          32s
pipelinecontroller              1        1        1           1          32s
jenkins-x-sonatype-nexus        1        1        1           0          32s

==> v1beta1/CustomResourceDefinition
NAME                  AGE
pipelines.jenkins.io  32s
runs.jenkins.io       32s

==> v1/Pod(related)
NAME                                             READY  STATUS   RESTARTS  AGE
jenkins-x-chartmuseum-d48f95c7d-ttfn9            1/1    Running  0         32s
jenkins-x-docker-registry-794dcb698-h2tcv        1/1    Running  0         32s
jenkins-x-heapster-76df87bd59-47g4j              2/2    Running  0         28s
jenkins-77f44b4c76-hzc96                         0/1    Running  0         32s
jenkins-x-kubernetes-dashboard-6b87d747b7-p8krt  1/1    Running  0         32s
jenkins-x-mongodb-6c84f866dc-j24j8               1/1    Running  0         32s
jenkins-x-monocular-api-6548574cb5-xvb8d         0/1    Running  1         32s
jenkins-x-monocular-prerender-6d8897856-ghzsf    1/1    Running  0         32s
jenkins-x-monocular-ui-7dccdf6c9b-g2fls          0/1    Running  0         32s
pipelinecontroller-5dbc96bbd4-ft5zp              1/1    Running  0         31s
jenkins-x-sonatype-nexus-5b976bd546-cdk2d        0/1    Running  0         31s

==> v1/Secret
NAME                              TYPE    DATA  AGE
jenkins-x-docker-registry-secret  Opaque  1     33s
jenkins                           Opaque  2     33s
jenkins-x-mongodb                 Opaque  2     33s
jenkins-docker-cfg                Opaque  1     33s
jenkins-git-credentials           Opaque  1     33s
jenkins-hub-api-token             Opaque  1     33s
jenkins-git-ssh                   Opaque  2     33s
jx-basic-auth                     Opaque  1     33s
jenkins-release-gpg               Opaque  4     33s
jenkins-maven-settings            Opaque  1     33s
jenkins-npm-token                 Opaque  1     33s
jenkins-ssh-config                Opaque  1     33s

==> v1/ConfigMap
NAME                              DATA  AGE
jenkins-x-docker-registry-config  1     33s
jenkins                           3     33s
jenkins-tests                     1     33s
jenkins-x-monocular-api-config    1     33s
jenkins-x-monocular-ui-config     1     33s
jenkins-x-monocular-ui-vhost      1     33s

==> v1/PersistentVolumeClaim
NAME                      STATUS  VOLUME                                    CAPACITY  ACCESS MODES  STORAGECLASS  AGE
jenkins-x-chartmuseum     Bound   pvc-3893faeb-0c74-11e8-8c25-26c3e276d95b  8Gi       RWO           standard      33s
jenkins                   Bound   pvc-3894f09c-0c74-11e8-8c25-26c3e276d95b  3Gi       RWO           standard      33s
jenkins-x-mongodb         Bound   pvc-3896f70c-0c74-11e8-8c25-26c3e276d95b  8Gi       RWO           standard      33s
jenkins-x-sonatype-nexus  Bound   pvc-3897bfec-0c74-11e8-8c25-26c3e276d95b  4Gi       RWO           standard      32s


waiting for install to be ready, if this is the first time then it will take a while to download images
Error: Looks like "http://admin:admin@chartmuseum.default.192.168.64.7.nip.io" is not a valid chart repository or cannot be reached: Get http://admin:admin@chartmuseum.default.192.168.64.7.nip.io/index.yaml: dial tcp 192.168.64.7:80: getsockopt: connection refused
Error: Command failed  helm repo add releases http://admin:admin@chartmuseum.default.192.168.64.7.nip.io
```

**Comment by michaelneale** [comment]
I see this: 

```
Update Complete. ⎈ Happy Helming!⎈ 
helm install jenkins-x/jenkins-x-platform --name jenkins-x -f ./myvalues.yaml -f ./secrets.yaml --version 0.0.55 --values=/Users/michaelneale/.jx/gitSecrets.yaml --values=/Users/michaelneale/.jx/extraValues.yaml --namespace=default
Error: timed out waiting for the condition
make: *** [install] Error 1
Error: Command failed  make ARGS=--values=/Users/michaelneale/.jx/gitSecrets.yaml --values=/Users/michaelneale/.jx/extraValues.yaml --namespace=default install
error: exit status 2
```

on jx install (jx install can't be tried again, as "a release named jenkins-x already exists."

**Comment by jstrachan** [comment]
@michaelneale the `a release named jenkins-x already exists.` comes from helm; if you uninstall you should be able to install again. Sounds like it took too long to download the images (we should make the timeout configurable).

Does uninstall + install help?

**Comment by jstrachan** [comment]
@i386 that error looks like helm can't access `chartmuseum.default.192.168.64.7.nip.io` - I wonder if the code cannot accurately detect you're using minikube. As the installer says:
```
You can now configure a wildcard DNS pointing to the new loadbalancer address 192.168.64.7
? Domain 192.168.64.7.nip.io
``` 
which is normally reserved for doing GKE/AKS/EKS type installs.

I wonder if this helps:
```
jx uninstall
jx install --provider minikube
```
The way the auto-detection of minikube goes is its looking at the current kubernetes context name and checking for minikube.
I wonder what the output of this is:
```
 kubectl config current-context
```

If you've tried using, say, GKE or switching kubectl contexts to different clusters your ~/.kube/config file can get in a mess leaving it with an odd context name which `jx` won't be able to detect easily if you're using minikube.

One final thought; whats the output of 
```
kubectl get node -oyaml
```
Maybe the code that tries to detect minikube could use something in there too to make really sure it is on minikube?

**Comment by jstrachan** [comment]
@i386 ah looking at the code it does output `You can now configure a wildcard DNS pointing to the new loadbalancer address 192.168.64.7
` on minikube afterall - so its maybe nothing to do with the minikube detection.

So probably more likely that your machine can't access the ingress URLs (e.g. `chartmuseum.default.192.168.64.7.nip.io`).

e.g. if you type:

```
kubectl get pod
```
Are all the pods running?

If so if you type:
```
jx open
```
you should see a list of URLs you can click such as jenkins. Can you open any of them? Jenkins + Nexus should work for example.

If you can't open those URLs it could be that something's messed up with the ingress controller on your minikube - or something wacky networking related?  It also could be 'nip.io' is down for you.

One work around - I added a command yesterday for days when 'nip.io' is down:
```
sudo jx create etc-hosts
```
Which adds to your /etc/hosts all the ingress URLs like `chartmuseum.default.192.168.64.7.nip.io` pointing at your local ip `minikube ip` so that your machine should be able to access all the URLs which end in `*.default.192.168.64.7.nip.io`. I wonder if that helps?

**Comment by i386** [comment]
@jstrachan Yep, looks like I didn't have ingress enabled.  Perhaps we should warn if the ingress isn't setup or available? I've updated the docs as part of #220

**Comment by jstrachan** [comment]
ah thanks. I wonder if we should prompt the user if they'd like to do the `sudo jx create etc-hosts` as its a nice way to handle things whether or not ingress is running, whether or not you have internet & whether or not nip.io is working that day etc
